pkg/autoid_service/autoid.go

```diff
@@ -453,6 +453,8 @@ func (s *Service) allocAutoID(ctx context.Context, req *autoid.AutoIDRequest) (*
 	})
 
 	val := s.getAlloc(req.DbID, req.TblID, req.IsUnsigned)
+	val.Lock()
```

Comment:
What is the expected performance impact since we're using locks in wider scope.

Could you provide some kind of benchmark result, if possible?

Reply:
For the `AllocID` API, I think it's almost the same with before, because **the old code did use lock here too** ...  the only difference is `req.N == 0` code branch which means a "get current autoid value" query.

For the `Rebase` API, the performance could be worse than before, but this query is rare comparing to the `AllocID` API, so the performance is not as sensitive.

From my experience data, simple insert into SQL query on AUTO_ID_CACHE=1 tables workload on my local machine get 140K QPS, and the `AllocID` API itself performs better than that.   @bb7133